central/sensor/service/connection/sensorevents.go

```diff
@@ -84,6 +91,10 @@ func (s *sensorEventHandler) addMultiplexed(ctx context.Context, msg *central.Ms
 			// NodeInventory because the two should not dedupe between themselves.
 			msg.DedupeKey = fmt.Sprintf("NodeInventory:%s", msg.GetDedupeKey())
 		default:
+			if evt.Event.GetResource() == nil {
```

Comment:
Is it possible for Event to be nil (and thus should be checked earlier)?

Reply:
We are accessing `Resource` already in `switch evt.Event.Resource.(type)` - I think, we need to do the nil check earlier.

Reply:
Yeah I'll add a check for event == nil

Reply:
Note, that on master a panic will actually be caught so this category of issue would be less pressing

Reply:
I'm usually a fan of using the proto's Getters, but it's not used anywhere else in the function, and there isn't a *need* for it since we know evt.Event is not nil